rust-src/id/src/id_proof_types.rs

```diff
@@ -0,0 +1,380 @@
+        self.statements
+            .push(AtomicStatement::RevealAttribute { statement });
+        self
+    }
+
+    pub fn in_range(
+        mut self,
+        tag: AttributeTag,
+        lower: AttributeType,
+        upper: AttributeType,
+    ) -> Self {
+        let statement = AttributeInRangeStatement {
+            attribute_tag: tag,
+            lower,
+            upper,
+            _phantom: PhantomData::default(),
+        };
+        self.statements
+            .push(AtomicStatement::AttributeInRange { statement });
+        self
+    }
+
+    pub fn member_of(mut self, tag: AttributeTag, set: BTreeSet<AttributeType>) -> Self {
+        let statement = AttributeInSetStatement {
+            attribute_tag: tag,
+            set,
+            _phantom: PhantomData::default(),
+        };
+        self.statements
+            .push(AtomicStatement::AttributeInSet { statement });
+        self
+    }
+
+    pub fn not_member_of(mut self, tag: AttributeTag, set: BTreeSet<AttributeType>) -> Self {
+        let statement = AttributeNotInSetStatement {
+            attribute_tag: tag,
+            set,
+            _phantom: PhantomData::default(),
+        };
+        self.statements
+            .push(AtomicStatement::AttributeNotInSet { statement });
+        self
+    }
+
+    pub fn residence_in(mut self, set: BTreeSet<AttributeType>) -> Self {
+        let statement = AttributeInSetStatement {
+            attribute_tag: AttributeTag(4u8), // country of residence
+            set,
+            _phantom: PhantomData::default(),
+        };
+        self.statements
+            .push(AtomicStatement::AttributeInSet { statement });
+        self
+    }
+
+    pub fn residence_not_in(mut self, set: BTreeSet<AttributeType>) -> Self {
+        let statement = AttributeNotInSetStatement {
+            attribute_tag: AttributeTag(4u8), // country of residence
+            set,
+            _phantom: PhantomData::default(),
+        };
+        self.statements
+            .push(AtomicStatement::AttributeNotInSet { statement });
+        self
+    }
+
+    pub fn document_issuer(&mut self, set: BTreeSet<AttributeType>) -> &mut Self {
+        let statement = AttributeInSetStatement {
+            attribute_tag: AttributeTag(4u8),
+            set,
+            _phantom: PhantomData::default(),
+        };
+        self.statements
+            .push(AtomicStatement::AttributeInSet { statement });
+        self
+    }
+}
+
+/// A proof about a credential on an account
+#[derive(Debug, Clone, SerdeSerialize, SerdeDeserialize)]
+#[serde(bound(
+    serialize = "C: Curve, AttributeType: Attribute<C::Scalar> + SerdeSerialize",
+    deserialize = "C: Curve, AttributeType: Attribute<C::Scalar> + SerdeDeserialize<'de>"
+))]
+pub struct Proof<C: Curve, AttributeType: Attribute<C::Scalar>> {
+    pub account:    AccountAddress,
+    #[serde(serialize_with = "base16_encode", deserialize_with = "base16_decode")]
+    pub credential: CredId<C>,
```

Comment:
Please document the meaning of these fields. It is a bit unclear without knowing the context.

Reply:
I will remove these fields since they were actually not used. (The verifier knows the account and credential already, since they are in the statement, and the prover knows them too when constructing the proof from the statement).